src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,755 @@
+  }
+  if ((strcmp(emethod, "ARKODE_LSRK_SSP_S_2") == 0) ||
+      (strcmp(emethod, "ARKODE_LSRK_SSP_S_3") == 0) ||
+      (strcmp(emethod, "ARKODE_LSRK_SSP_10_4") == 0))
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__,
+                    __FILE__, "Invalid method option: Call LSRKStepCreateSTS to create an STS method first.");
+  }
+
+  arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                  "Unknown method type");
+
+  return ARK_ILL_INPUT;
+}
+
+int LSRKStepSetSSPMethodByName(void* arkode_mem, const char* emethod)
+{
+  if ((strcmp(emethod, "ARKODE_LSRK_RKC_2") == 0) ||
+      (strcmp(emethod, "ARKODE_LSRK_RKL_2") == 0))
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__,
+                    __FILE__, "Invalid method option: Call LSRKStepCreateSSP to create an SSP method first.");
+  }
+  if (strcmp(emethod, "ARKODE_LSRK_SSP_S_2") == 0)
+  {
+    return LSRKStepSetSSPMethod(arkode_mem, ARKODE_LSRK_SSP_S_2);
+  }
+  if (strcmp(emethod, "ARKODE_LSRK_SSP_S_3") == 0)
+  {
+    return LSRKStepSetSSPMethod(arkode_mem, ARKODE_LSRK_SSP_S_3);
+  }
+  if (strcmp(emethod, "ARKODE_LSRK_SSP_10_4") == 0)
+  {
+    return LSRKStepSetSSPMethod(arkode_mem, ARKODE_LSRK_SSP_10_4);
+  }
+
+  arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                  "Unknown method type");
+
+  return ARK_ILL_INPUT;
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigFn specifies the dom_eig function.
+  Specifies the dominant eigenvalue approximation routine to be used for determining
+  the number of stages that will be used by either the RKC or RKL methods.
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFn(void* arkode_mem, ARKDomEigFn dom_eig)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return retval; }
+
+  /* set the dom_eig routine pointer, and update relevant flags */
+  if (dom_eig != NULL)
+  {
+    step_mem->dom_eig_fn = dom_eig;
+
+    return ARK_SUCCESS;
+  }
+  else
+  {
+    step_mem->dom_eig_fn = NULL;
+
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Internal dom_eig is not supported yet!");
+    return ARK_ILL_INPUT;
+  }
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigFrequency sets dom_eig computation frequency -
+  Dominated Eigenvalue is recomputed after "nsteps" successful steps.
+
+  nsteps = 0 refers to constant dominant eigenvalue
+  nsteps < 0 resets the default value 25 and sets nonconstant dominant eigenvalue
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFrequency(void* arkode_mem, int nsteps)
```

Comment:
Update to `long int` to be consistent with other similar options
```suggestion
int LSRKStepSetDomEigFrequency(void* arkode_mem, long int nsteps)
```